<launch>

	<env name="PX4_SIM_MODEL" value="solo" />
		<!-- 估计器参数选择 可选ekf2_vision和ekf2_gps-->
		<!-- ekf2_gps 使用GPS作为定位来源， ekf2_vision 使用外部输入（gazebo真值、slam等）作为定位来源-->
		<!-- 参看 ~/prometheus_px4/ROMFS/px4fmu_common/init.d-posix/rcS 中的修改内容 -->
    <env name="PX4_ESTIMATOR" value="ekf2" />
		<!-- 仿真速度因子 1.0代表与真实时间同步，大于1加快仿真速度，小于1则减慢 （电脑性能较差，可选择减小该参数）-->
	<env name="PX4_SIM_SPEED_FACTOR" value="1.0" />

    <arg name="agent_id" default="1"/>
    <arg name="uav_name" default="uav1"/>
    <arg name="setmode_timeout" default="2"/>
    <arg name="stop_mode" default="2"/>
    <arg name="swarm_node_name" default="swarm_control_uav1"/>
    <arg name="control_node_name" default="uav_control_uav1"/>
    <arg name="sim_mode" default="false"/>
    <arg name="swarm" default="true"/>
    <arg name="sdf_path" default="$(find prometheus_gazebo)/amov_models/Multisolo/uav1/uav1.sdf"/>
    <arg name="control/controller_flag" default="0"/>
    <arg name="control/enable_external_control" default="false"/>
    <arg name="control/Takeoff_height" default="1.0"/>
    <arg name="takeoff_height" default="1.0"/>
    <arg name="control/Disarm_height" default="0"/>
    <arg name="control/Land_speed" default="0"/>
    <arg name="control/location_source" default="0"/>
    <arg name="geo_fence/x_min" default="0"/>
    <arg name="geo_fence/x_max" default="0"/>
    <arg name="geo_fence/y_min" default="0"/>
    <arg name="geo_fence/y_max" default="0"/>
    <arg name="geo_fence/z_min" default="0"/>
    <arg name="geo_fence/z_max" default="0"/>
    <arg name="uav_init_x" default="0"/>
    <arg name="uav_init_y" default="0"/>

    <!--swarm controll 节点-->
    <node pkg="prometheus_mission" type="swarm_control_node" name="$(arg swarm_node_name)" output="screen">
        <param name="agent_id" value="$(arg agent_id)"/>
        <param name="takeoff_height" value="$(arg takeoff_height)"/>
        <param name="setmode_timeout" value="$(arg setmode_timeout)"/>
        <param name="stop_mode" value="$(arg stop_mode)"/>
    </node>

    <!--uav control 节点-->
    <node pkg="prometheus_uav_control" type="uav_control_main" name="$(arg control_node_name)" output="screen">
        <param name="uav_id" value="$(arg agent_id)"/>
        <param name="sim_mode" value="$(arg sim_mode)"/>
        <param name="swarm" value="$(arg swarm)"/>
        <param name="control/controller_flag" value="$(arg control/controller_flag)"/>
        <param name="control/enable_external_control" value="$(arg control/enable_external_control)"/>
        <param name="control/Takeoff_height" value="$(arg control/Takeoff_height)"/>
        <param name="control/Disarm_height" value="$(arg control/Disarm_height)"/>
        <param name="control/Land_speed" value="$(arg control/Land_speed)"/>
        <param name="control/location_source" value="$(arg control/location_source)"/>
        <param name="geo_fence/x_min" value="$(arg geo_fence/x_min)"/>
        <param name="geo_fence/x_max" value="$(arg geo_fence/x_max)"/>
        <param name="geo_fence/y_min" value="$(arg geo_fence/y_min)"/>
        <param name="geo_fence/y_max" value="$(arg geo_fence/y_max)"/>
        <param name="geo_fence/z_min" value="$(arg geo_fence/z_min)"/>
        <param name="geo_fence/z_max" value="$(arg geo_fence/z_max)"/>
    </node>

    <!--无人机仿真-->
	<include file="$(find prometheus_mission)/launch/sitl_basic.launch">
		<arg name="uav_id" value="$(arg agent_id)"/>
		<arg name="sdf" value="$(arg sdf_path)"/>
		<arg name="model" value="$(arg uav_name)"/>
		<arg name="uav_init_x" value="$(arg uav_init_x)"/>
		<arg name="uav_init_y" value="$(arg uav_init_y)"/>
		<arg name="uav_init_z" value="0.15"/>
		<arg name="uav_init_yaw" value="0.0"/>
	</include>

</launch>
